### PR TITLE
Fixes overlay tracks to allow for multiple extent values and 4-tuples.

### PR DESCRIPTION
**Pull request by mildewey**
## Description

What was changed in this pull request?
A) Overlay tracks were augmented to allow for extent to draw multiple ranges. This was an intended feature that was not implemented properly. B) It was also augmented to allow for drawing x and y ranges differently for the "same" extent. This may or may not have been intended initially, but allows for flexibility in how overlay tracks are drawn.

Why is it necessary?
The alternative was to make a large number of overlay tracks and configure each identically. This seemed obnoxious and counterintuitive.

Fixes #___

## Checklist

- [x] Unit tests added or updated
- [x] Documentation added or updated
- [x] Example added or updated
- [x] Screenshot for visual changes (e.g. new tracks)
- [x] Updated CHANGELOG.md
higlassoverlay

### CODE REVIEW

**Comment by pkerpedjiev** [comment]
Could you please fill out the PR request template? The changes look fine and they seem to work when I tried them.

File: app/scripts/OverlayTrack.js [hunk 1]
```diff
@@ -21,38 +49,19 @@ class OverlayTrack extends PixiTrack {
...
-          + this._yScale(this.options.extent[0][0]);
-        const width = position.width;
-        const height = this._yScale(this.options.extent[0][1])
-          - yPos
-          + position.top
-          + this.position[1];
-
-        graphics.drawRect(xPos, yPos, width, height);
+      if (op.orientation === '1d-vertical' || op.orientation === '2d') {
+        this.options.extent.forEach(extent => this.drawVerticalOverlay(
```
Comment (flekschas) [inline app/scripts/OverlayTrack.js]: Will `this.options.extent.forEach` break if `extent` is configured incorrectly? A fix could be to test at the beginning whether extent is an array, i.e., `this.extent = this.options.extent || []`

Comment (flekschas) [reply app/scripts/OverlayTrack.js]: (Same in the above loop)

File: app/scripts/OverlayTrack.js [hunk 2]
```diff
@@ -11,6 +11,34 @@ class OverlayTrack extends PixiTrack {
     this.drawnRects = {};
   }
 
+  drawHorizontalOverlay(graphics, position, extent) {
```
Comment (flekschas) [inline app/scripts/OverlayTrack.js]:
```suggestion
  drawHorizontalOverlay(graphics, position, extent = [0, 1]) {
```

File: app/scripts/OverlayTrack.js [hunk 3]
```diff
@@ -11,6 +11,34 @@ class OverlayTrack extends PixiTrack {
...
+    const height = position.height;
+    const width = this._xScale(extent[1])
+      - xPos
+      + position.left
+      + this.position[0];
+
+    graphics.drawRect(xPos, yPos, width, height);
+  }
+
+  drawVerticalOverlay(graphics, position, extent) {
```
Comment (flekschas) [inline app/scripts/OverlayTrack.js]:
```suggestion
  drawVerticalOverlay(graphics, position, extent = [0, 1]) {
```

**Comment by flekschas** [comment]
Looks fine. I've just left a few minor comments regarding robustness. Also, please add or extend:

- the existing example
- CHANGELOG.md

File: app/scripts/OverlayTrack.js (hunk 2 above)
Comment (mildewey) [reply app/scripts/OverlayTrack.js]: Resolved by checking if extent exists and is long enough and doing nothing if either is bad.

File: app/scripts/OverlayTrack.js (hunk 1 above)
Comment (mildewey) [reply app/scripts/OverlayTrack.js]: Fixed by making sure we have extent as an array.

**Pull request merged**